File: src/main/java/com/ism/dreamssellersv2/service/AuthService.java
package com.ism.dreamssellersv2.service;

import com.ism.model.LoginDTO;
import com.ism.model.LoginResponseDTO;
import com.ism.model.RegisterDTO;

public interface AuthService {
    LoginResponseDTO register(RegisterDTO registerDTO);
    LoginResponseDTO login(LoginDTO loginDTO);
}

File: frontend/src/app/services/auth.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {LoginDTO, LoginResponseDTO, RegisterDTO} from '../models/auth.model';
import {environment} from '../../environments/environment';

@Injectable({
  providedIn: 'root',
})
export class AuthService {
  private apiUrl = `${environment.apiUrl}/auth`;

  constructor(private http: HttpClient) {}

  // Login
  login(credentials: LoginDTO): Observable<LoginResponseDTO> {
    return this.http.post<LoginResponseDTO>(`${this.apiUrl}/login`, credentials);
  }

  // Register
  register(user: RegisterDTO): Observable<LoginResponseDTO> {
    return this.http.post<LoginResponseDTO>(`${this.apiUrl}/register`, user);
  }

}
